underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystem.java

```diff
@@ -126,6 +126,22 @@ public GCSUnderFileSystem(AlluxioURI uri) throws ServiceException {
     mBucketMode = GCSUtils.translateBucketAcl(acl, mAccountOwnerId);
   }
 
+  protected GCSUnderFileSystem(AlluxioURI uri,
```

Comment:
@goodluckbot we can make `this(...)` the first line by using helper methods to supply the more complicated default field values, e.g.
`this(createGoogleStorageService(), uri.getHost(), PathUtils.normalizePath(Constants.HEADER_GCS + uri.getHost(), PATH_SEPARATOR), etc)`

Alternately, we could change the other constructor to a factory method, e.g. 
```java
public GCSUnderFileSystem create(AlluxioURI uri) {
   ...
   return new GCSUnderFileSystem(...)
}
```